crypto/sql_store.go

```diff
@@ -734,6 +734,29 @@ func (store *SQLCryptoStore) FilterTrackedUsers(ctx context.Context, users []id.
 	return dbutil.NewRowIter(rows, dbutil.ScanSingleColumn[id.UserID]).AsList()
 }
 
+// MarkTrackedUsersOutdated flags that the device list for given users are outdated.
+func (store *SQLCryptoStore) MarkTrackedUsersOutdated(ctx context.Context, users []id.UserID) error {
+	return store.DB.DoTxn(ctx, nil, func(ctx context.Context) error {
+		for _, userID := range users {
+			_, err := store.DB.Exec(ctx, "INSERT OR REPLACE INTO crypto_tracked_user (user_id, devices_outdated) VALUES ($1, TRUE)", userID)
```

Comment:
This should be `UPDATE`

Reply:
Actually it doesn't work the intended way if it's an UPDATE. We want to flag users as tracked and outdated at the same time if the first time we see an encrypted event from them in a sync but we don't know their devices at all we want to flag and track them to do the device fetching.

Reply:
Maybe if we rename it to `MarkUsersTrackedAndOutdated` so it is explicit what it does?